.azure-pipelines/stage/publish.yml

```diff
@@ -402,18 +404,24 @@ jobs:
   pool:
     vmImage: $(agentUbuntu)
   steps:
-  - task: DownloadBuildArtifacts@0
-    inputs:
-      buildType: current
-      artifactName: "release.signed"
-      itemPattern: "release.signed/release.signed.tar.zst"
-      targetPath: $(Build.StagingDirectory)
   - template: ../bazel.yml
     parameters:
       ciTarget: publish
       cacheVersion: $(cacheKeyBazel)
       publishEnvoy: false
       publishTestResults: false
+      stepsPre:
+      - bash: |
+          set -e
+          if [[ "$BUILD_REASON" == "PullRequest" ]]; then
+              DOWNLOAD_PATH="$(git rev-parse HEAD | head -c7)"
+          else
+              DOWNLOAD_PATH="${SYSTEM_PULLREQUEST_PULLREQUESTNUMBER:-${BUILD_SOURCEBRANCHNAME}}"
+          fi
+          curl -sLO "https://storage.googleapis.com/${{ parameters.bucketGCP }}/${DOWNLOAD_PATH}/release/release.signed.tar.zst"
+          mkdir -p $(Build.StagingDirectory)/release.signed
+          cp -a release.signed.tar.zst $(Build.StagingDirectory)/release.signed
```

Comment:
nit: will move be better here?

Reply:
not sure it matters that much, but updated anyway